tests/check/check_signed_video_auth.c

```diff
@@ -1120,29 +1120,36 @@ START_TEST(detect_change_of_public_key)
   // |settings|; See signed_video_helpers.h.
 
   // Generate 2 GOPs
-  test_stream_t *list = create_signed_nalus("IPPIPP", settings[_i]);
-  test_stream_check_types(list, "IPPSIPP");
-
+  test_stream_t *list = create_signed_nalus("IPPIPPI", settings[_i]);
+  test_stream_check_types(list, "IPPSIPPSI");
+  test_stream_item_t *i_item = test_stream_item_remove(list, 9);
+  test_stream_item_check_type(i_item, 'I');
   // Generate another GOP from scratch
   // This will generate a new private key, hence transmit a different public key.
-  test_stream_t *list_with_new_public_key = create_signed_nalus_int("IPPPIPPI", settings[_i], true);
-  test_stream_check_types(list_with_new_public_key, "IPPPSIPPSI");
-
+  test_stream_t *list_with_new_public_key =
+      create_signed_nalus_int("IPIPIPPPIPPI", settings[_i], true);
+  test_stream_check_types(list_with_new_public_key, "IPSIPSIPPPSIPPSI");
+  int remove_items = 6;
+  while (remove_items--) {
```

Comment:
I think you should use `test_stream_pop(...)` instead of using a while loop. That function will remove the X first items and return a new list including them.